### PR TITLE
Profile dashboard notifications

### PR DESCRIPTION
**Pull request by mjoyce91**
Addresses #826 by adding Notifications to the user profile dashboard. Adds a reusable `BorderedList` component that can accept an array of React elements and place them inside `<li>` elements with a border (this will be used for the new Bid List since it follows the same style as notifications).

Dependent on #898.

### CODE REVIEW

File: src/Components/ProfileDashboard/CurrentUser/Status/Status.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import FontAwesome from 'react-fontawesome';
+
+const Status = () => (
+  <div className="usa-grid-full status">
+    <FontAwesome name="circle" /> Status: On-Post
```
Comment (burgwyn) [inline src/Components/ProfileDashboard/CurrentUser/Status/Status.jsx]: Should this be a prop?

File: src/Components/ProfileDashboard/Notifications/Notifications.jsx [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
+import Notification from './Notification';
 
-const Notifications = () => (
-  <div className="usa-grid-full">
-    Notifications
-  </div>
-);
+const Notifications = ({ notifications }) => {
+  const notificationArray = [];
+  // only show the first 3
```
Comment (burgwyn) [inline src/Components/ProfileDashboard/Notifications/Notifications.jsx]: Can we move this business rule somewhere else?  Are we testing with different values?

File: src/reducers/notifications.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import * as reducers from './notifications';
+
+describe('reducers', () => {
```
Comment (burgwyn) [inline src/reducers/notifications.test.js]: "notifications reducers"

File: src/reducers/assignment.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as reducers from './assignment';
+
+describe('reducers', () => {
```
Comment (burgwyn) [inline src/reducers/assignment.test.js]: "assignment reducers"

File: src/Components/ProfileDashboard/CurrentUser/EditProfile/EditProfile.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+const Status = () => (
```
[the remark this answers is not in the document]
Comment (mjoyce91) [reply src/Components/ProfileDashboard/CurrentUser/EditProfile/EditProfile.jsx]: 🍝

File: src/Components/ProfileDashboard/CurrentUser/Status/Status.jsx (hunk 1 above)
Comment (mjoyce91) [reply src/Components/ProfileDashboard/CurrentUser/Status/Status.jsx]: I purposefully left it static for now, waiting for us to flesh out some kind of status lifecycle.

File: src/Components/ProfileDashboard/Notifications/Notifications.jsx (hunk 2 above)
Comment (mjoyce91) [reply src/Components/ProfileDashboard/Notifications/Notifications.jsx]: I'm considering removing this requirement and simply rendering all notifications passed to the component. The API request already limits it to 3 with `?limit=3`.

**Pull request merged**